trade-finance-manager-api/src/v1/helpers/amendment.helpers.js

```diff
@@ -498,26 +498,41 @@ const calculateAmendmentDateTenor = async (coverEndDate, existingFacility) => {
   }
 };
 
-// populates tfmObject with date values
-const addLatestAmendmentDates = async (tfmObject, latestDate, facilityId) => {
-  // if there is a latest coverEndDate
-  if (latestDate?.coverEndDate) {
-    const { coverEndDate } = latestDate;
-    const existingFacility = await api.findOneFacility(facilityId);
+/**
+ * Populates the tfmObject with date values
+ * @param {}tfmObject
+ * @param latestCoverEndDateResponse
+ * @param latestFacilityEndDateDataResponse
+ * @param facilityId
+ * @returns {Object}
+ */
+const addLatestAmendmentDates = async (tfmObject, latestCoverEndDateResponse, latestFacilityEndDateDataResponse, facilityId) => {
+  const existingFacility = await api.findOneFacility(facilityId);
 
-    if (!existingFacility) {
-      return tfmObject;
-    }
-    // populates with coverEndDate and exposurePeriod
-    const newTfmObject = {
-      ...tfmObject,
-      coverEndDate: latestDate.coverEndDate,
+  if (!existingFacility) {
+    return tfmObject;
+  }
+
+  let newTfmObject = tfmObject;
+
+  if (latestCoverEndDateResponse?.coverEndDate) {
+    const { coverEndDate } = latestCoverEndDateResponse;
+    newTfmObject = {
+      ...newTfmObject,
+      coverEndDate,
       amendmentExposurePeriodInMonths: await calculateAmendmentDateTenor(coverEndDate, existingFacility),
     };
+  }
 
-    return newTfmObject;
+  if (latestFacilityEndDateDataResponse?.facilityEndDate) {
+    const { facilityEndDate } = latestFacilityEndDateDataResponse;
+    newTfmObject = {
+      ...newTfmObject,
+      facilityEndDate,
+    };
   }
-  return tfmObject;
+
+  return newTfmObject;
```

Comment:
I think this section could be made a lot more concise by doing something like the below
```typescript
const coverEndDate = latestCoverEndDateResponse?.coverEndDate;
const amendmentExposurePeriodInMonths = coverEndDate ? await calculateAmendmentDateTenor(coverEndDate, existingFacility) : undefined;
const facilityEndDate = latestFacilityEndDateDataResponse?.facilityEndDate;

return {
  ...tfmObject,
  coverEndDate,
  amendmentExposurePeriodInMonths,
  facilityEndDate,
};
```

What do you think? This is likely just a personal styling thing but I still thought it may be worth suggesting